### PR TITLE
Upstream 21.3.3.1 merge

### PR DESCRIPTION
**Pull request by jerboaa**
This is a simple merge from upstream `release/graal-vm/21.3` branch (which includes the upstream 21.3.3.1 changes. The intent is to later do a 21.3.4 release of mandrel for the October 2022 CPU update.

### CODE REVIEW

**Comment by jerboaa** [comment]
For the JDK 17-based Windows build we are hitting https://github.com/graalvm/mandrel/issues/419. So we'll need a backport of https://github.com/graalvm/mandrel/commit/0c88c7d5ded86bf566ba04660f6588c794131add to 21.3. Note that JDK 17 includes [JDK-6782021](https://bugs.openjdk.org/browse/JDK-6782021) earlier. In `17.0.5` which is this release.

File: regex/mx.regex/suite.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
   "name" : "regex",
 
   "version" : "21.3.3.1",
-  "release" : False,
+  "release" : True,
```
Comment (zakkak) [inline regex/mx.regex/suite.py]: I would avoid this kind of change. We can either do it in the merge commit or in a following commit.

**Review by zakkak — APPROVED** [review]
LGTM except for the release change

File: regex/mx.regex/suite.py (hunk 1 above)
Comment (jerboaa) [reply regex/mx.regex/suite.py]: Fair enough.

Comment (jerboaa) [reply regex/mx.regex/suite.py]: @zakkak On second thought, is it worth doing? The upstream branch has it set to true:
https://github.com/oracle/graal/blob/release/graal-vm/21.3/regex/mx.regex/suite.py#L47

 Could we do a version bump to `21.3.4.0` instead and set it back to false in a separate commit? Would that work for you?

**Comment by jerboaa** [comment]
> For the JDK 17-based Windows build we are hitting #419. So we'll need a backport of [0c88c7d](https://github.com/graalvm/mandrel/commit/0c88c7d5ded86bf566ba04660f6588c794131add) to 21.3. Note that JDK 17 includes [JDK-6782021](https://bugs.openjdk.org/browse/JDK-6782021) earlier. In `17.0.5` which is this release.

https://github.com/graalvm/mandrel/pull/430 adds it. Ideally after we've done the upstream merge and bumped to `21.3.4.0`.

**Comment by jerboaa** [comment]
This test failure:

```
 Caused by: sun.security.validator.ValidatorException: PKIX path validation failed: sun.security.validator.ValidatorException: TrustAnchor with subject "CN=*.badssl.com, O=BadSSL, L=San Francisco, ST=California, C=US" is not a CA certificate
	at sun.security.validator.PKIXValidator.doValidate(PKIXValidator.java:369)
	at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:275)
	at sun.security.validator.Validator.validate(Validator.java:264)
	at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
	at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:222)
	at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:129)
	at sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:638)
	... 57 more
Caused by: sun.security.validator.ValidatorException: TrustAnchor with subject "CN=*.badssl.com, O=BadSSL, L=San Francisco, ST=California, C=US" is not a CA certificate
	at sun.security.validator.PKIXValidator.verifyTrustAnchor(PKIXValidator.java:393)
	at sun.security.validator.PKIXValidator.toArray(PKIXValidator.java:333)
	at sun.security.validator.PKIXValidator.doValidate(PKIXValidator.java:366)
	... 63 more

Error:  Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.659 s <<< FAILURE! - in io.quarkus.it.rest.client.selfsigned.ExternalSelfSignedITCase
Error:  io.quarkus.it.rest.client.selfsigned.ExternalSelfSignedITCase.should_accept_self_signed_certs_java_url  Time elapsed: 1.275 s  <<< FAILURE!
```

@zakkak @Karm Is it expected?
See https://github.com/graalvm/mandrel/actions/runs/3227864470/jobs/5283632758#step:11:1329

[edit] AFAIK it didn't happen for the 21.3.3 merge here: https://github.com/graalvm/mandrel/actions/runs/2707392039/jobs/4231149852#step:11:1252

File: regex/mx.regex/suite.py (hunk 1 above)
Comment (zakkak) [reply regex/mx.regex/suite.py]: Yes, that's what I was referring to in "or in a following commit." :)

**Comment by zakkak** [comment]
> @zakkak @Karm Is it expected?

Yes, those are unrelated to the merge. See https://github.com/graalvm/mandrel/actions/runs/3224038072/jobs/5274878427#step:11:2242

File: regex/mx.regex/suite.py (hunk 1 above)
Comment (jerboaa) [reply regex/mx.regex/suite.py]: NVM. I've re-done the merge and let it to `False` now.

**Comment by jerboaa** [comment]
> > @zakkak @Karm Is it expected?
> 
> Yes, those are unrelated to the merge. See https://github.com/graalvm/mandrel/actions/runs/3224038072/jobs/5274878427#step:11:2242

OK, thanks!

**Comment by jerboaa** [comment]
Diff stat now is (all changes irrelevant for Mandrel):

```
 common.json                                                                                                         |  12 ++++-----
 compiler/ci_common/gate.hocon                                                                                       |   2 +-
 tools/CHANGELOG.md                                                                                                  |   3 +++
 tools/src/com.oracle.truffle.tools.agentscript/src/com/oracle/truffle/tools/agentscript/impl/InsightFilter.java     |  26 +++++++++++-------
 tools/src/com.oracle.truffle.tools.agentscript/src/com/oracle/truffle/tools/agentscript/impl/InsightInstrument.java |   6 ++---
 tools/src/com.oracle.truffle.tools.agentscript/src/com/oracle/truffle/tools/agentscript/impl/InsightPerContext.java |  15 +++++------
 tools/src/com.oracle.truffle.tools.agentscript/src/com/oracle/truffle/tools/agentscript/impl/InsightPerSource.java  |  71 +++++++++++++++++++++++++++++++++++++++++--------
 tools/src/org.graalvm.tools.insight.heap/src/org/graalvm/tools/insight/heap/instrument/HeapGenerator.java           |  10 ++++---
 tools/src/org.graalvm.tools.insight.test/src/org/graalvm/tools/insight/test/heap/HeapObjectStreamTest.java          |  29 +++++++++++---------
 tools/src/org.graalvm.tools.insight.test/src/org/graalvm/tools/insight/test/heap/HeapObjectTest.java                |  22 ++++++++++-----
 truffle/external_repos/simplelanguage/pom.xml                                                                       |   2 +-
 truffle/external_repos/simplelanguage/sl                                                                            |   2 +-
 truffle/external_repos/simpletool/pom.xml                                                                           |   2 +-
 truffle/src/com.oracle.truffle.api.test/src/com/oracle/truffle/api/test/host/GR39979.java                           | 127 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 truffle/src/com.oracle.truffle.api.test/src/com/oracle/truffle/api/test/polyglot/TruffleFileTest.java               |  40 +++++++++++++++++++++++++---
 truffle/src/com.oracle.truffle.polyglot/src/com/oracle/truffle/polyglot/FileSystems.java                            |  17 +++++++++++-
 truffle/src/com.oracle.truffle.polyglot/src/com/oracle/truffle/polyglot/PolyglotExecuteNode.java                    |  18 +++++++------
 vm/mx.vm/suite.py                                                                                                   |   6 ++---
 18 files changed, 327 insertions(+), 83 deletions(-)
```

**Review by zakkak — APPROVED** [review]

**Pull request merged**